Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -476,55 +480,78 @@ void pfPatcherWorker::Run()
 
 void pfPatcherWorker::ProcessFile()
 {
+    plFileName clientDir = plFileSystem::GetCurrentAppPath().StripFileName();
+
     do {
         NetCliFileManifestEntry& entry = fQueuedFiles.front();
+        fQueuedFiles.pop_front();
 
         // eap sucks
-        plFileName clName = ST::string::from_wchar(entry.clientName);
+        plFileName clName = plFileName(ST::string::from_wchar(entry.clientName)).Normalize();
         ST::string dlName = ST::string::from_wchar(entry.downloadName);
 
+        // Only operate on files if they are contained within the client directory...
+        plFileName clAbsPath = clName.AbsolutePath();
+        if (!clAbsPath.AsString().starts_with(clientDir.AsString())) {
```

Comment:
The ostensible reason for this fix was to prevent potential conflicts in errant PRPs due to the way that they are loaded.  I see no such cause for concern with the asset files (except, possibly, the .pak archives).

It's a separate issue to be culling old or unused files simply to reduce clutter, and that's not one I think is worth the added security risk.